test/throughput/test_collector_profile.py

```diff
@@ -0,0 +1,129 @@
+import gym
+import numpy as np
+import pytest
+from gym.spaces.discrete import Discrete
+from gym.utils import seeding
+
+from tianshou.data import Batch, Collector, ReplayBuffer
+from tianshou.env import VectorEnv
+from tianshou.policy import BasePolicy
+
+
+class SimpleEnv(gym.Env):
+    """A simplest example of self-defined env, used to minimize
+    data collect time and profile collector."""
+
+    def __init__(self):
+        self.action_space = Discrete(200)
+        self._fake_data = np.ones((10, 10, 1))
+        self.seed(0)
+        self.reset()
+
+    def reset(self):
+        self._index = 0
+        self.done = np.random.randint(3, high=200)
+        return {'observable': np.zeros((10, 10, 1)),
+                'hidden': self._index}
+
+    def step(self, action):
+        if self._index == self.done:
+            raise ValueError('step after done !!!')
+        self._index += 1
+        return {'observable': self._fake_data, 'hidden': self._index}, -1, \
+            self._index == self.done, {}
+
+    def seed(self, seed=None):
+        self.np_random, seed = seeding.np_random(seed)
+        return [seed]
+
+
+class SimplePolicy(BasePolicy):
+    """A simplest example of self-defined policy, used
+    to minimize data collect time."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def learn(self, batch, **kwargs):
+        return super().learn(batch, **kwargs)
+
+    def forward(self, batch, state=None, **kwargs):
+        return Batch(act=np.array([30]*len(batch)), state=None, logits=None)
+
+
+@pytest.fixture(scope="module")
+def data():
+    np.random.seed(0)
+    env = SimpleEnv()
+    env.seed(0)
+    envs = VectorEnv(
+        [lambda: SimpleEnv() for _ in range(100)])
+    envs.seed(np.random.randint(1000, size=100).tolist())
+    buffer = ReplayBuffer(50000)
+    policy = SimplePolicy()
+    collector = Collector(policy, env, ReplayBuffer(50000))
+    collector_multi = Collector(policy, envs, ReplayBuffer(50000))
+    return{
+        "env": env,
+        "policy": policy,
+        "envs": envs,
+        "buffer": buffer,
+        "collector": collector,
+        "collector_multi": collector_multi
+    }
+
+
+def test_init(data):
```

Comment:
Is it necessary?

Reply:
But I mean the `__init__` test.

Reply:
Well, for now this seems not necessary because __init__ really doesn't take much time, but in my opinion, we cannot guarantee that profiling __init__ is of no use in the future. I simply leave it there as a template and let it be consistent with other profile test scripts. I could delete this if you really consider this unnecessary. What do you think?